pkg/generators/markers.go

```diff
@@ -36,7 +36,87 @@ type CommentTags struct {
 	// Default  any  `json:"default,omitempty"`
 }
 
-func ParseCommentTags(comments []string, prefix string) (CommentTags, error) {
+// validates the parameters in the comment tags.
+func (c CommentTags) Validate(t types.Type) error {
+
+	reolvedType := resolveAliasAndPtrType(&t)
+
+	switch reolvedType.Kind {
+	case types.Builtin:
+		if c.MaxItems != nil || c.MinItems != nil || c.UniqueItems || c.MaxProperties != nil || c.MinProperties != nil || c.AdditionalProperties != nil {
+			return fmt.Errorf("invalid validation for builtin type")
+		}
+		switch t.Name.Name {
+		case "string":
+			if c.Minimum != nil || c.Maximum != nil || c.MultipleOf != nil {
+				return fmt.Errorf("invalid validation for string type")
+			}
+		case "boolean":
+			if c.Minimum != nil || c.Maximum != nil || c.MultipleOf != nil || c.MinLength != nil || c.MaxLength != nil || c.Pattern != "" {
+				return fmt.Errorf("invalid validation for boolean type")
+			}
+		default:
+			if c.MinLength != nil || c.MaxLength != nil || c.Pattern != "" {
+				return fmt.Errorf("invalid validation for %s type", t.Name)
+			}
+		}
+	case types.Slice, types.Array:
+		if c.Minimum != nil || c.Maximum != nil || c.MultipleOf != nil || c.MinProperties != nil || c.MaxProperties != nil || c.MinLength != nil || c.MaxLength != nil || c.Pattern != "" {
+			return fmt.Errorf("invalid validation for %s type", t.Name)
+		}
+	case types.Map:
+		if c.Minimum != nil || c.Maximum != nil || c.MultipleOf != nil || c.MinItems != nil || c.MaxItems != nil || c.MinLength != nil || c.MaxLength != nil || c.Pattern != "" {
+			return fmt.Errorf("invalid validation for %s type", t.Name)
+		}
+	}
```

Comment:
For the enforcement of which validation can go on which type i think youd be better served by not repeating conditions, and also more specific error messages:

```go
reolvedType := resolveAliasAndPtrType(&t)
typeString, _ := openapi.OpenAPITypeFormat(reolvedType.String()) // will be empty for complicated types
isArray := resolvedType.Kind == types.Slice || resolvedType.Kind == types.Array
isMap := resolveType.Kind == types.Map
isString := typeString == "string"
isInt := typeString == "integer"
isFloat := typeString == "number"
isBool := typeString == "boolean"

if c.MaxItems != nil && !isArray {
    allErrors = append(allErrors, fmt.Errorf("maxItems may only be specified on arrays"))
}

...
```